Sources/web3swift/EthereumABI/ABIDecoding.swift

```diff
@@ -137,7 +137,12 @@ extension ABIDecoder {
                         let (v, c) = decodeSignleType(type: subType, data: dataSlice, pointer: subpointer)
                         guard let valueUnwrapped = v, let consumedUnwrapped = c else {break}
                         toReturn.append(valueUnwrapped)
-                        subpointer = subpointer + consumedUnwrapped
+                        if (subType.isStatic) {
+                            subpointer = subpointer + consumedUnwrapped
+                        }
+                        else {
+                            subpointer = consumedUnwrapped // need to go by nextElementPointer
```

Comment:
Breaks decoding of arrays that contain any type that is not static. 
That would be `string`, `dynamicBytes`, `array` and `tuple` that contain non-static types.

@izakpavel Could you please explain what you meant by this comment?
> // need to go by nextElementPointer

Thanks!

Reply:
Here is an example with values that I'm working with. 
What I'm doing is encoding two arrays and using that result as an argument for the decoding function.

This is the first array to be encoded.
`ABI.Element.ParameterType` is `array(bytes(32))`:
```
[0] - 0xdf30dba06db6a30e65354d9a64c609861f089545ca58c6b4dbe31a5f338cb0e3
[1] - 0x4b80742d0000000082ac000022300fd00e3402df12db7a0a054fae50f0bf91d7
[2] - 0x027545a85de73f6a6ec5ce14736fa7eb00000000000000000000000000000002
```

This is the second array to encoded. 
`ABI.Element.ParameterType` is `array(dynamicBytes)` even though in this specific example all values are 32 bytes long:
```
[0] - 0x0000000000000000000000000000000000000000000000000000000000000003
[1] - 0x00000000000000000000000000000000000000000000000000000000000003fe
[2] - 0x22300fd00e3402df12db7a0a054fae50f0bf91d7000000000000000000000000
```

And this is how they are decoded:
Array with `bytes(32)`:
```
[0] - 0xdf30dba06db6a30e65354d9a64c609861f089545ca58c6b4dbe31a5f338cb0e3
[1] - 0x4b80742d0000000082ac000022300fd00e3402df12db7a0a054fae50f0bf91d7
[2] - 0x027545a85de73f6a6ec5ce14736fa7eb00000000000000000000000000000002
```

Array with dynamicBytes:
```
[0] - 0x0000000000000000000000000000000000000000000000000000000000000003
[1] - 0x00000000000000000000000000000000000000000000000000000000000003fe
[2] - 0x00000000000000000000000000000000000000000000000000000000000003fe
```

Reply:
@BaldyAsh Please, take a look at this issue. It feels like reverting line 144 back to 
```
subpointer = subpointer + consumedUnwrapped
```
would fix the issue.

Note: tested the suggestion above in a fork of web3swift and it does indeed fix the issue. 
I'll test later with `array(tuple)` type as @izakpavel had issues with it.

Reply:
`array(tuple)` does break indeed if line 144 is reverted to its previous logic.